unison-cli/src/Unison/Codebase/Editor/HandleInput/Update.hs

```diff
@@ -189,12 +201,324 @@ handleUpdate input optionalPatch requestedNames = do
         & Path.resolve @_ @_ @Path.Absolute currentPath'
         & tShow
 
+getSlurpResultForUpdate :: Set Name -> Cli r (SlurpResult Symbol)
+getSlurpResultForUpdate requestedNames = do
+  (slurp, termRefToNames, nameToTermRefs) <- do
+    slurpCheckNames <- Branch.toNames <$> Cli.getCurrentBranch0
```

Comment:
I notice this same set of names is being constructed twice, both in the call location and again inside the function; maybe pass them in instead?